dispatch/route.go

```diff
@@ -88,14 +88,28 @@ func NewRoute(cr *config.Route, parent *Route) *Route {
 	}
 
 	// Build matchers.
-	var matchers types.Matchers
+	var matchers labels.Matchers
 
 	for ln, lv := range cr.Match {
```

Comment:
I'd comment on this `for` loop and the next that this handles the deprecated case.